packages/js-drive/lib/masternodes/manageMasternodesIdentities.js

```diff
@@ -0,0 +1,207 @@
+const { hash } = require('@dashevo/dpp/lib/util/hash');
+const IdentityPublicKey = require('@dashevo/dpp/lib/identity/IdentityPublicKey');
+const Transaction = require('@dashevo/dashcore-lib/lib/transaction');
+
+/**
+ *
+ * @param {DashPlatformProtocol} transactionalDpp
+ * @param {DriveStateRepository|CachedStateRepositoryDecorator} stateRepository
+ * @param {createIdentity} createIdentity
+ * @return {manageMasternodesIdentities}
+ */
+function manageMasternodesIdentitiesFactory(
+  transactionalDpp,
+  stateRepository,
+  createIdentity,
+) {
+  /**
+   * @typedef manageMasternodesIdentities
+   * @param {SimplifiedMasternodeList} simplifiedMasternodeList
+   * @param {number} latestRequestedHeight
+   * @param {number} coreHeight
+   * @return Promise<void>
+   */
+  async function manageMasternodesIdentities(
+    simplifiedMasternodeList,
+    latestRequestedHeight,
+    coreHeight,
+  ) {
+    const documentsToCreate = [];
+    const documentsToDelete = [];
+
+    const previousMNList = simplifiedMasternodeList.getStore()
+      .getSMLbyHeight(latestRequestedHeight)
```

Comment:
You need to compare with all diffs in store because previous last is just change from previous block, but you need the whole state.